frontend/server/src/main/java/org/pytorch/serve/wlm/SequenceBatchAggregator.java

```diff
@@ -0,0 +1,213 @@
+package org.pytorch.serve.wlm;
+
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.pytorch.serve.job.Job;
+import org.pytorch.serve.job.JobGroup;
+import org.pytorch.serve.util.messages.BaseModelRequest;
+import org.pytorch.serve.util.messages.ModelWorkerResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SequenceBatchAggregator extends BatchAggregator {
+
+    private static final Logger logger = LoggerFactory.getLogger(SequenceBatchAggregator.class);
+    private ExecutorService pollExecutors;
+    private LinkedBlockingDeque<String> eventJobGroupIds;
+    private LinkedBlockingDeque<Job> jobQueue;
+    private Thread eventDispatcher;
+    private AtomicBoolean isPollJobGroup;
+    private LinkedList<String> currentJobGroupIds;
+
+    public SequenceBatchAggregator(Model model) {
+        super(model);
+        this.currentJobGroupIds = new LinkedList<>();
+        this.pollExecutors = Executors.newFixedThreadPool(model.getBatchSize() + 1);
+        this.jobQueue = new LinkedBlockingDeque<>();
+        this.isPollJobGroup = new AtomicBoolean(false);
+        this.eventJobGroupIds = new LinkedBlockingDeque<>();
+        this.eventJobGroupIds.add("");
+        this.eventDispatcher = new Thread(new EventDispatcher());
+        this.eventDispatcher.start();
+    }
+
+    public void startEventDispatcher() {
+        this.eventDispatcher.start();
+    }
+
+    public void stopEventDispatcher() {
+        this.eventDispatcher.interrupt();
+    }
+
+    private void pollJobGroup() throws InterruptedException {
+        if (isPollJobGroup.getAndSet(true)) {
+            return;
+        }
+        LinkedHashSet<String> tmpJobGroups = new LinkedHashSet<>();
+        String jobGroupId = model.getPendingJobGroups().poll(Long.MAX_VALUE, TimeUnit.MILLISECONDS);
+        if (jobGroupId != null) {
+            addJobGroup(jobGroupId);
+
+            int quota = model.getPendingJobGroups().size() / model.getMaxWorkers();
```

Comment:
As we do not know about the other workers load here it would be better to set quota to max(quota, remaining_capacity)